src/test/java/org/opensearch/ad/transport/ADJobParameterTransportActionTests.java

```diff
@@ -0,0 +1,83 @@
+package org.opensearch.ad.transport;
+
+import static org.mockito.Mockito.*;
+
+import java.io.IOException;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.ad.TestHelpers;
+import org.opensearch.ad.util.RestHandlerUtils;
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.xcontent.*;
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.extensions.action.ExtensionActionRequest;
+import org.opensearch.extensions.action.ExtensionActionResponse;
+import org.opensearch.jobscheduler.spi.JobDocVersion;
+import org.opensearch.jobscheduler.transport.ExtensionJobActionRequest;
+import org.opensearch.jobscheduler.transport.JobParameterRequest;
+import org.opensearch.tasks.Task;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.transport.TransportService;
+
+public class ADJobParameterTransportActionTests extends OpenSearchIntegTestCase {
+
+    private ADJobParameterTransportAction action;
+
+    private Task task;
+
+    private ActionListener<ExtensionActionResponse> response;
+
+    private ExtensionActionRequest extensionActionRequest;
+
+    private JobDocVersion jobDocVersion;
+
+    private XContentParser parser;
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+
+        action = new ADJobParameterTransportAction(mock(TransportService.class), mock(ActionFilters.class), xContentRegistry());
+        task = mock(Task.class);
+        jobDocVersion = new JobDocVersion(1L, 1L, 1L);
+        response = new ActionListener<>() {
+
+            @Override
+            public void onResponse(ExtensionActionResponse extensionActionResponse) {
+                Assert.assertTrue(true);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                Assert.assertTrue(true);
+            }
+        };
+    }
+
+    @Test
+    public void testJobParameterTransportAction() throws IOException {
+        XContentBuilder content = TestHelpers.randomXContent();
+        setExtensionActionRequest(content);
+        action.doExecute(task, extensionActionRequest, response);
+    }
+
+    @Test
+    public void testJobParameterTransportActionWithZeroJobSource() throws IOException {
+        XContentBuilder content = JsonXContent.contentBuilder().startObject().endObject();
+        setExtensionActionRequest(content);
+        action.doExecute(task, extensionActionRequest, response);
+    }
+
+    private void setExtensionActionRequest(XContentBuilder content) throws IOException {
+        parser = XContentHelper
```

Comment:
This is probably identical code to other `createParser()` implementations in AnomalyDetectorJobRunner.java L435-437 and ADJobRunnerTransportAction.java L100.  Curious why it's different here?

Reply:
Just following DRY principle here in both the test of LIne 63 and 70. In Line 65 and Line 72 the content has different data in them so just made one common method to follow DRY principle